### ISSUE TITLE
PSCi re-export handling needs some work

### ISSUE THREAD

**Issue by garyb** [purescript/purescript#2080]
Currently re-exported values will not be autocompleted in psci. I'm not exactly sure what the behaviour here should be, I guess it should behave as if they were defined within the module the import is being attempted for.

This can probably be solved by some use of the `Env` from `Sugar.Names`, but I'm also considering doing some work so re-exports will be elaborated as values in the module's exports lists directly, as I think all round this will result in easier re-export handling.

**Comment by paf31** [comment]
PSCi is starting to have more and more in common with `psc-ide` and I've been wondering if it makes sense to try to combine them. Note `psc-ide` already does some work to figure out what to do with reexports.

Also, `:load` will act a lot like `psc-ide` once #2071 is merged.

**Comment by hdgarrood** [comment]
Might it make sense to postpone this for an 0.9.x release (where x > 0)?

**Comment by garyb** [comment]
Possibly, yeah. I'm about to dive back into the names stuff, so will see if some of the changes I have planned there make it easier thought.

**Comment by hdgarrood** [comment]
Resolved by #3231
